Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditorSettings.cpp

```diff
@@ -0,0 +1,126 @@
+            return pathFilter.starts_with(path.substr(0, pathFilter.size()));
+        };
+        dumperSettings.m_jsonPointerPrefix = m_settingsRegistryBasePath;
+
+        auto outcome = m_settingsRegistrar.SaveSettingsToFile(m_fullSettingsFilepath, dumperSettings, m_settingsRegistryBasePath);
+        if (!outcome.IsSuccess())
+        {
+            AZ_Warning("DocumentPropertyEditorSettings", false, outcome.GetError().c_str());
+        }
+    }
+
+    bool DocumentPropertyEditorSettings::LoadExpanderStates()
+    {
+        auto loadOutcome = m_settingsRegistrar.LoadSettingsFromFile(m_fullSettingsFilepath);
+        if (loadOutcome.IsSuccess())
+        {
+            auto getOutcome = m_settingsRegistrar.GetObjectSettings(this, m_fullSettingsRegistryPath);
+            if (!getOutcome.IsSuccess())
+            {
+                AZ_Warning("DocumentPropertyEditorSettings", false, getOutcome.GetError().c_str());
+                return false;
+            }
+        }
+        else
+        {
+            AZ_Warning("DocumentPropertyEditorSettings", false, loadOutcome.GetError().c_str());
+            return false;
+        }
+
+        return true;
+    }
+
+    void DocumentPropertyEditorSettings::SaveAndCleanExpanderStates()
+    {
+        // Attempt to remove old settings from registry if the local state was successfully cleaned
+        // This way we save and dump to file the most accurate expander settings
+        if (m_cleanExpanderStateCallback && m_cleanExpanderStateCallback(m_expandedElementStates))
+        {
+            if (!m_settingsRegistrar.RemoveSettingFromRegistry(m_fullSettingsRegistryPath))
+            {
+                AZ_Warning("DocumentPropertyEditorSettings", false,
+                    "Failed to clean registry state %s before saving", m_fullSettingsRegistryPath.c_str());
+            }
+        }
+
+        if (!m_expandedElementStates.empty())
+        {
+            SaveExpanderStates();
+        }
+    }
+
+    void DocumentPropertyEditorSettings::SetExpanderStateForRow(const AZ::Dom::Path& rowPath, bool isExpanded)
+    {      
+        m_expandedElementStates[rowPath.ToString()] = isExpanded;
+    }
+
+    bool DocumentPropertyEditorSettings::GetExpanderStateForRow(const AZ::Dom::Path& rowPath)
+    {
+        AZStd::string_view strPath = rowPath.ToString();
+        if (m_expandedElementStates.contains(strPath))
+        {
+            return m_expandedElementStates[strPath];
+        }
+        return false;
+    }
+
+    bool DocumentPropertyEditorSettings::HasSavedExpanderStateForRow(const AZ::Dom::Path& rowPath) const
+    {
+        return m_expandedElementStates.contains(rowPath.ToString());
+    }
+
+    void DocumentPropertyEditorSettings::RemoveExpanderStateForRow(const AZ::Dom::Path& rowPath)
+    {
+        const AZStd::string& strPath = rowPath.ToString();
+        if (m_expandedElementStates.contains(strPath))
+        {
+            m_expandedElementStates.erase(strPath);
+        }
```

Comment:
The contains check isn't needed here.
AZStd::unordered_map erase can just be called directly. It already checks if the key is contained in the map.
```suggestion
        m_expandedElementStates.erase(strPath);
```

Reply:
Done.